sdks/go/pkg/beam/runners/dataflow/dataflow.go

```diff
@@ -282,24 +285,51 @@ func getJobOptions(ctx context.Context) (*dataflowlib.JobOptions, error) {
 	hooks.SerializeHooksToOptions()
 
 	experiments := jobopts.GetExperiments()
-	// Always use runner v2, unless set already.
-	var v2set, portaSubmission bool
+	// Ensure that we enable the same set of experiments across all SDKs
+	// for runner v2.
+	var fnApiSet, v2set, uwSet, portaSubmission, seSet, wsSet bool
 	for _, e := range experiments {
-		if strings.Contains(e, "use_runner_v2") || strings.Contains(e, "use_unified_worker") {
+		if strings.Contains(e, "beam_fn_api") {
+			fnApiSet = true
+		}
+		if strings.Contains(e, "use_runner_v2") {
 			v2set = true
 		}
+		if strings.Contains(e, "use_unified_worker") {
+			uwSet = true
+		}
 		if strings.Contains(e, "use_portable_job_submission") {
 			portaSubmission = true
 		}
+		if strings.Contains(e, "disable_runner_v2") || strings.Contains(e, "disable_runner_v2_until_2023") || strings.Contains(e, "disable_prime_runner_v2") {
+			return nil, errors.New("disabling runner v2 is now longer supported as of 2.45.0+")
```

Comment:
Maybe add a little more information like: `Detected one of the following experiments: disable_runner_v2 | disable_runner_v2_until_2023 | disable_prime_runner_v2. Disabling runner v2 is no longer supported as of Beam version 2.45.0+`

(also, typo nit: `now` -> `no`)

Reply:
done